src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditionsImpl.hpp

```diff
@@ -0,0 +1,1447 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <tuple>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/DataOnSlice.hpp"
+#include "DataStructures/LeviCivitaIterator.hpp"
+#include "DataStructures/SliceVariables.hpp"
+#include "DataStructures/TempBuffer.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Domain/IndexToSliceAt.hpp"
+#include "Domain/Tags.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditionsHelpers.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Characteristics.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Constraints.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/FluxCommunicationTypes.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/MortarHelpers.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+/// \cond
+namespace Tags {
+template <typename Tag>
+struct Magnitude;
+}  // namespace Tags
+// IWYU pragma: no_forward_declare db::DataBox
+/// \endcond
+
+namespace GeneralizedHarmonic {
+namespace Actions {
+
+namespace BoundaryConditions_detail {}  // namespace BoundaryConditions_detail
+
+namespace BoundaryConditions_detail {
+enum class VSpacetimeMetricBcMethod {
+  Freezing,
+  ConstraintPreservingBjorhus,
+  ConstraintPreservingDirichlet,
+  Unknown
+};
+enum class VZeroBcMethod {
+  Freezing,
+  ConstraintPreservingBjorhus,
+  ConstraintPreservingDirichlet,
+  ConstraintPreservingRealDirichlet,
+  Unknown
+};
+enum class VPlusBcMethod { Freezing, Unknown };
+enum class VMinusBcMethod {
+  Freezing,
+  ConstraintPreservingBjorhus,
+  ConstraintPreservingPhysicalBjorhus,
+  Unknown
+};
+
+template <size_t VolumeDim>
+using all_local_vars = tmpl::list<
```

Comment:
Update: maybe a comment saying you'll introduce aliases below for each quantity would help the reader

Reply:
Overhauled the computation of intermediate variables. Now, the computation and storage are both handled by a single standalone computer class and we use human-readable aliases everywhere instead of indexed names.